app/controllers/districts_controller.rb

```diff
@@ -13,19 +25,28 @@ def new
 
   def create
     @district = District.new(district_params)
+    @district.user = current_user
     if @district.save
+      redirect_to @district
       flash[:success] = "District added successfully"
-      redirect_to districts_path
     else
       @district.errors.any?
       flash[:notice] = @district.errors.full_messages.join(", ")
       render :new
     end
   end
-end
 
-private
+  protected
+
+  def district_params
+    params.require(:district).permit(:name, :description)
+  end
+
+  def authorize_user
+
+    if !user_signed_in?
+      raise ActionController::RoutingError.new('Not Found')
+    end
```

Comment:
For readability, it's better to write `if !` as `unless`. Also, this action seems somewhat redundant; returning a routing error instead of simply relying on Devise's `authenticate_user` method is a high-security feature (verging on paranoid), and I'm not sure I understand its purpose here.